doozerlib/assembly.py

```diff
@@ -184,16 +190,19 @@ def assembly_basis_event(releases_config: Model, assembly: str) -> typing.Option
     return assembly_basis_event(releases_config, target_assembly.basis.assembly)
 
 
-def assembly_config_finalize(releases_config: Model, assembly: str, rpm_metas, ordered_image_metas):
+def assembly_basis(releases_config: Model, assembly: str) -> typing.Optional[Model]:
     """
-    Some metadata cannot be finalized until all metadata is read in by doozer. This method
-    uses that interpreted metadata set to go through and adjust assembly information
-    within it.
-    :param releases_config: The releases.yml Model
-    :param assembly: The name of the assembly to apply
-    :param rpm_metas: A list of rpm metadata to update relative to the assembly.
-    :param ordered_image_metas: A list of image metadata to update relative to the assembly.
-    :return: N/A. Metas are updated in-place. Only call during runtime.initialize.
+    :param releases_config: The content of releases.yml in Model form.
+    :param assembly: The name of the assembly to assess
+    Returns the basis dict for a given assembly. If the assembly has no basis,
+    None is returned.
     """
+    if not assembly or not isinstance(releases_config, Model):
+        return None
+
     _check_recursion(releases_config, assembly)
-    pass
+    target_assembly = releases_config.releases[assembly].assembly
+    if target_assembly.basis:
+        return target_assembly.basis
```

Comment:
If `target_assembly` has a basis, it will not inherit the basis from its parent assembly. This is insistent with how we handle `assembly_basis_event` and seems confusing.

Reply:
Yep, you are right.